libpod/container_internal.go

```diff
@@ -1268,7 +1268,10 @@ func (c *Container) start() error {
 		}
 	}
 
-	if c.config.HealthCheckConfig != nil {
+	// Check if healthcheck is not nil and --no-healthcheck option is not set.
+	// If --no-healthcheck is set Test will be always set to `[NONE]` so no need
+	// to update status in such case.
+	if c.config.HealthCheckConfig != nil && !(len(c.config.HealthCheckConfig.Test) == 1 && c.config.HealthCheckConfig.Test[0] == "NONE") {
```

Comment:
Could len(c.config.HealthCheckConfig.Test)  > 1 and still have a value somewhere of "none"?  If so what should we do there?

Reply:
yes, but that check should probably be much much further up the stack imho

Reply:
@baude is correct I think we already have a conflict check for that.